core/modules/filters/equals.js

```diff
@@ -0,0 +1,63 @@
+/*\
+title: $:/core/modules/filters/equals.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator for comparing text for equality
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.equals = function(source,operator,options) {
+	var results = [],
+		fieldname = (operator.suffix || operator.operator || "title").toLowerCase();
+	if(operator.prefix === "!") {
+		if(operator.regexp) {
+			source(function(tiddler,title) {
+
+				var text = title ;
+					if(text !== null && !operator.regexp.exec(text)) {
+						results.push(title);
+					}
+			});
+		} else {
+			source(function(tiddler,title) {
+
+				var text = title ;
+					if(text !== null && text !== operator.operand) {
+						results.push(title);
+					}
+			});
+		}
+	} else {
+		if(operator.regexp) {
+			source(function(tiddler,title) {
+				
+				var text = title ;
+				if(text !== null && !!operator.regexp.exec(text)) {
+					results.push(title);
+				}
+				
+			});
+		} else {
+			source(function(tiddler,title) {
+				
+				var text = title ;
+				if(text !== null && text === operator.operand) {
+					results.push(title);
+				}
+				
+			});
+		}
+	}
+	return results;
+};
```

Comment:
You could really tighten this up using a predicate map:

```js
exports.equals = function(source,operator,options) {
	var results = [];
	var predicates = {
		negateRegexp: function (text, operator) {
			return text !== null && !operator.regexp.exec(text);
		},
		nagateOperand: function (text, operator) {
			return (text !== null && text !== operator.operand);
		},
		affirmRegexp: function (text, operator) {
			return text !== null && !!operator.regexp.exec(text);
		},
		affirmOperand: function (text, operator) {
			return text !== null && text === operator.operand;
		}
	};
	var predicateKey = (operator.prefix === '!' ? 'negate' : 'affirm') +
		(operator.regexp ? 'Regexp' : 'Operand');
	source(function (tiddler, title) {
		if (predicates[predicateKey](title, operator)) {
			results.push(title);
		}
	});
	return results;
};
```

Reply:
My strategy is to take existing code and modify it as gently as possible. That way there is little chance of introducing new errors.  While there may be more sophisticated approaches, complete re-writes are like little time bombs in my experience.   If you want to write new code and submit it, then Jeremy can decide which approach he prefers.

Reply:
Maybe I missunderstood. The diff was all green so this is a **new** file and hence not a refactor.

Reply:
It's technically new. It's a rewrite of the field operator, but short-circuiting the bits that require the presence of an actual tiddler.

Reply:
The code for the core filter operators is generally "unrolled" to pull  conditionals out of the loop. Of course, such optimisations may not be needed in these days of super-optimising JS runtimes, but I'd want to make some measurements before changing the approach.